### DISCUSSION TITLE
needed some help

### DISCUSSION THREAD

**Discussion by iirsh** [community/community#145449] [New to GitHub]
is it possible to make a campus navigation where the map is offline, and is it possible to  make the app using html and css? if so can someone help me?

**Comment by CyberSphinxxx** [comment]
Hello @iirsh

Yes, it's definitely possible to create a campus navigation app with an offline map!

While HTML and CSS can help you design the app, you'll need JavaScript for the map functionality. You can use libraries like **Leaflet.js** or **OpenLayers** to display maps offline by saving the map tiles locally on the user's device. JavaScript can also help you track the user's location and show points of interest on the map. 

So, HTML and CSS are great for the layout, but JavaScript will be key for implementing offline map features.

**Comment by enzodelcompare** [comment]
Hi @iirsh!

Yes, it's possible to create an offline campus navigation app using HTML, CSS, and JavaScript. You can use:

- [Leaflet.js](https://leafletjs.com/): For interactive maps.
- Offline Map Tiles: Store map tiles locally (e.g., MBTiles format or pre-downloaded PNG tiles).
- JavaScript: For route calculations and navigation logic (e.g., using Dijkstra or A* algorithms).
- PWA (Progressive Web App): To enable offline functionality and caching.
- Storage Options: Use localStorage or IndexedDB for offline data storage.

Here’s a basic structure:

```
<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" href="https://unpkg.com/leaflet/dist/leaflet.css" />
    <style>#map { height: 100vh; }</style>
</head>
<body>
    <div id="map"></div>
    <script src="https://unpkg.com/leaflet/dist/leaflet.js"></script>
    <script>
        var map = L.map('map').setView([latitude, longitude], 15);
        L.tileLayer('path_to_tiles/{z}/{x}/{y}.png', {
            maxZoom: 18,
            minZoom: 12,
        }).addTo(map);
        L.marker([latitude, longitude]).addTo(map).bindPopup('You are here').openPopup();
    </script>
</body>
</html>
```

**Comment by ghostinhershell** [comment]
Hi @iirsh 👋🏾 

Have you tried using Copilot to build this project with you? If you haven't gotten a chance to try it out, [you can get started for free here](https://github.com/features/copilot?ef_id=_k_Cj0KCQiAkoe9BhDYARIsAH85cDPmrsgpkB8GuK6T03GBUMSlrUOgb1fJeHR9zoid1sOHTvumSVp_QcYaAp8uEALw_wcB_k_&OCID=AIDcmmb150vbv1_SEM__k_Cj0KCQiAkoe9BhDYARIsAH85cDPmrsgpkB8GuK6T03GBUMSlrUOgb1fJeHR9zoid1sOHTvumSVp_QcYaAp8uEALw_wcB_k_&gad_source=1&gclid=Cj0KCQiAkoe9BhDYARIsAH85cDPmrsgpkB8GuK6T03GBUMSlrUOgb1fJeHR9zoid1sOHTvumSVp_QcYaAp8uEALw_wcB).
